ui/src/alerting/components/notifications/StatusLevels.tsx

```diff
@@ -0,0 +1,71 @@
+// Libraries
+import React, {FC, useContext} from 'react'
+
+// Components
+import {
+  ComponentSpacer,
+  TextBlock,
+  FlexDirection,
+  ComponentSize,
+} from '@influxdata/clockface'
+import LevelsDropdown from 'src/alerting/components/notifications/LevelsDropdown'
+import StatusChangeDropdown from 'src/alerting/components/notifications/StatusChangeDropdown'
+import {NewRuleDispatch} from './NewRuleOverlay'
+
+// Types
+import {StatusRuleItem} from 'src/types'
+
+interface Props {
+  status: StatusRuleItem
+}
+
+const StatusLevels: FC<Props> = ({status}) => {
+  const {currentLevel, previousLevel} = status.value
+  const dispatch = useContext(NewRuleDispatch)
+
+  const onClickLevel = (type, level) => {
+    const value = {
+      ...status.value,
+      [type]: {
+        ...status.value[type],
+        level,
+      },
+    }
+
+    dispatch({
+      type: 'UPDATE_STATUS_RULES',
+      statusRule: {
+        ...status,
+        value,
+      },
+    })
+  }
+
+  return (
+    <ComponentSpacer direction={FlexDirection.Row} margin={ComponentSize.Small}>
+      <TextBlock text="When status" />
+      <ComponentSpacer.FlexChild grow={0} basis={140}>
```

Comment:
I wish this `ComponentSpacer` stuff didn't exist. We've adopted all the verbosity of a CSS-in-JS solution with none of the power.

Reply:
What if it was just renamed to <FlexBox />?

Reply:
The thing that gets me about it is that it causes us to add a ton more markup / divs everywhere. This looks really noisy when viewing the JSX or the DOM via devtools. Like in the hypothetical scenario:

```tsx
<Nav className="nav">
  <ComponentSpacer direction={FlexDirection.Column}>
    <ComponentSpacer.FlexChild grow={0} shrink={0} basis={50}>
      <NavItem className="nav-item" title="foo" />
    </ComponentSpacer.FlexChild>
  </ComponentSpacer>
</Nav>
```

Could just be

```tsx
<Nav className="nav">
  <NavItem className="nav-item" title="foo" />
</Nav>
```

with some corresponding CSS on `.nav` and `.nav-item`. You still have to define the the styling, but it's separate and once it's done you don't have to think about it or parse through all the `ComponentSpacer` noise to work with the component. 

This is the same thing that bugs me about CSS-in-JS libs like emotion or styled-components. Without a lot of care and strong conventions, you end up adding a lot more noise to your components. It also feels like we are defining a duplicate but less powerful API for something that is already so simple to achieve (putting flexbox styles on an element).

ofc this is all stuff I think should be hashed out systematically via clockface, not in this PR. Just commenting to share what I've been thinking about our current approach